### ISSUE TITLE
Advballsocket isn't pasted correctly when copied after dupe has moved.

### ISSUE THREAD

**Issue by thegrb93** [wiremod/advdupe2#212]

**Comment by UltimateByte** [comment]
Hey @thegrb93 Didn't know you were involved in addons development as well. Didn't I see you first in LinuxGSM community?

I've got players complaining about this issue since our (late but recent) update of advdupe2 on the server.
Do you know which (stable) version I can use to avoid this issue?

PS: If you need to test, server is game.terageek.org:27026
Screwball (Owner rank) will know more about the issue than I do.

**Comment by Exilphir** [comment]
So hey, it's me Screwball that @UltimateByte was talking about, we also got trouble with the last update of advdupe2 with the Advanced Ballsocket, it doesn't really require the build to move for it to bug, the advanced ballsocket can undo itself when there's too many of it, by exemple, i finish a side of ballsocket on a vehicles and they all work then i start to work on the other side, finish it, and some of the ballsocket just ... broke, like if they were never done.

**Comment by thegrb93** [comment]
@Exilphir the problem you're describing doesn't sound related to this issue. You're sure the issue started happening after updating advdupe2? I've seen this issue too, but it seems like a garry's mod issue rather than advdupe2.

If you can confirm it was updating advdupe2 that caused it, try an earlier version and see if it works.
Try this version for example, https://github.com/wiremod/advdupe2/tree/20063e65190509f5ebc763afb2604d4f30f1d220

If you can find a version where the issue stops occurring then that would help greatly.

**Comment by thegrb93** [comment]
To pick a version, go here https://github.com/wiremod/advdupe2/commits/master and find one you want to try and then click the <> button on the right.

**Comment by Exilphir** [comment]
Yeah it was after a update that the issues has been spotted, we will try some other build

**Comment by UltimateByte** [comment]
For now I can only say we had the late 2018 version before and didn't encounter the issue. So we have to try versions between then and now until we find the problematic version (and therefore the problematic commit).

**Comment by thegrb93** [comment]
Exactly

**Comment by thegrb93** [comment]
@Exilphir @UltimateByte any news?

**Comment by Exilphir** [comment]
I got quite buzy the last few but i found out the issues with the help of someone, apparently it's since the new sorting options that force to have under 100 constraints, it messed up a lot of the constraints even with the option disabled

**Comment by thegrb93** [comment]
The reason for that was found to be due to the wire clutch

**Comment by UltimateByte** [comment]
So in the end, is this a bug or a feature ? :o))

**Comment by thegrb93** [comment]
It's the wire clutch, it throws constraints out of order and causes constraint systems to form when they shouldn't. It's not an advdupe2 problem.

**Comment by thegrb93** [comment]
With the constraint sorting option disabled, you should get the same behavior as the old duplicator, is that not the case?

**Comment by Exilphir** [comment]
Even with the option disabled, the constraints are messed up

**Comment by thegrb93** [comment]
Do you have a contraption I can use to reproduce the issue that doesn't have wire clutches

**Comment by thegrb93** [comment]
The adv-ballsocket issue I believe is because advballsocket constraints depend on the orientation of the entity with respect to the world, so the entities need to be rotated to their world rotation at the time of creation of the constraint rather than just local to each other.
